geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java

```diff
@@ -472,12 +478,20 @@ public boolean waitUntilFlushed(String indexName, String regionPath, long timeou
         new WaitUntilFlushedFunctionContext(indexName, timeout, unit);
     Execution execution = FunctionService.onRegion(dataRegion);
     ResultCollector rs = execution.setArguments(context).execute(WaitUntilFlushedFunction.ID);
-    List<Boolean> results = (List<Boolean>) rs.getResult();
-    for (Boolean oneResult : results) {
-      if (oneResult == false) {
+    List<Object> results = (List<Object>) rs.getResult();
+    if (results != null) {
+      if (results.get(0) instanceof IllegalStateException) {
```

Comment:
Instead of checking types and silently handling this exception, it's probably better to just throw it and not have to modify this portion of code.